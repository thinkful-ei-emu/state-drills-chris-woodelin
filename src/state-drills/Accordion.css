.active, button:hover {
    background-color: #002444; 
  }

  button {
      padding: 40px;
      width:50%;
      border: none;
      font-size: 1em;
      color: #ffffff;
      background-color: lightgray;
  }

  li {
      text-align: center;
      list-style: none;
  }

  .content{
      background: black;
      color: white;
      width: 50%;
      margin: 0 auto;
      
  }
