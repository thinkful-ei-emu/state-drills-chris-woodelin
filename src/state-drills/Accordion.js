import React from 'react';

class Accordion extends React.Component {
    static defaultProps = { sections: [] };

    state = {
        currentSectionIndex: 0
    }

    renderButtons() {
        return this.props.sections.map((section, index) => (
            <li>
              <button key={index} onClick={() => this.handleButtonClick(index)}>
              {section.title}
              </button>
              <div style={{display: this.state.currentSectionIndex === index ? "block" : "none"}} className='content'>
                {section.content}
              </div>
            </li>
        ))
    }

    handleButtonClick(index) {
        this.setState({ currentSectionIndex: index })
    }

    render() {
        
        return(
            <div>
                {this.renderButtons()}
            </div>
        )
    }
}

export default Accordion;