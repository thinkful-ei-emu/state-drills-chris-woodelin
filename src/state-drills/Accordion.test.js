import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import Accordion from './Accordion';

describe(`Accordion Component`, () => {
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the first tab by default', () => {
    const tree = renderer.create(<Accordion sections={[
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
      ]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders empty given no sections', () => {
    const wrapper = shallow(<Accordion />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the first tab by default', () => {
    const wrapper = shallow(<Accordion sections={[
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
      ]} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})