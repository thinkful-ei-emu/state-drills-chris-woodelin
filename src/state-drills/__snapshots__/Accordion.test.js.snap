// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component renders empty given no sections 1`] = `<div />`;

exports[`Accordion Component renders the first tab by default 1`] = `
<div>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <div
      className="content"
      style={
        Object {
          "display": "block",
        }
      }
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </div>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <div
      className="content"
      style={
        Object {
          "display": "none",
        }
      }
    >
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </div>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <div
      className="content"
      style={
        Object {
          "display": "none",
        }
      }
    >
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </div>
  </li>
</div>
`;

exports[`Accordion Component renders the first tab by default 2`] = `
<div>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <div
      className="content"
      style={
        Object {
          "display": "block",
        }
      }
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </div>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <div
      className="content"
      style={
        Object {
          "display": "none",
        }
      }
    >
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </div>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <div
      className="content"
      style={
        Object {
          "display": "none",
        }
      }
    >
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </div>
  </li>
</div>
`;
